agent-tooling/src/main/java/io/opentelemetry/auto/typed/span/HttpClientTypedSpan.java

```diff
@@ -0,0 +1,20 @@
+package io.opentelemetry.auto.typed.span;
+
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+
+public abstract class HttpClientTypedSpan<T extends HttpClientTypedSpan, REQUEST, RESPONSE>
+    extends ClientTypedSpan<T, REQUEST, RESPONSE> {
+
+  public HttpClientTypedSpan(Tracer tracer, Span delegate) {
+    super(tracer, delegate);
+  }
+
+  public T onRequest(REQUEST request) {
+    tracer.getHttpTextFormat().inject(getContext(), request, getSetter());
+    return self();
+  }
+
+  protected abstract HttpTextFormat.Setter<REQUEST> getSetter();
```

Comment:
What do you think about putting this method in the typed tracer?

Reply:
done.